src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -146,6 +146,7 @@ public function step( $node_to_process = self::PROCESS_NEXT_NODE ) {
 		}
 
 		$this->current_bookmark = $this->bookmark_tag();
+		defined( 'DEBUG' ) && printf( "Encountering \e[31m{$this->get_tag()}\e[m Element at \e[35m{$this->current_bookmark}\e[m\n" );
```

Comment:
1. This is really elegant
2. It also brings bad memories of `mysql_query() or die()` :D

Reply:
I've been trying to think if there's a way we can keep some debugging info in there without ruining the source code. Continuing to stew on it, which is why I didn't erase this yet.

Reply:
I think of it as extended documentation. It could probably replace a few comments.